packages/controls/src/controls/rich-text/v2/rich-text.ts

```diff
@@ -0,0 +1,166 @@
+import { v4 as uuid } from 'uuid'
+import { z } from 'zod'
+
+import { type CopyContext } from '../../../context'
+import { type IntrospectionTarget } from '../../../introspection'
+import { safeParse, type ParseResult } from '../../../lib/zod'
+import { ControlDefinition, type SchemaType } from '../../definition'
+import { ControlInstance } from '../../instance'
+
+import { DTOSchema, isRichTextDTO, richTextDTOtoDAO } from '../dto'
+import * as Slate from '../slate'
+
+import { copyNodes } from './copy'
+import { introspectNodes } from './introspection'
+import { RichTextPluginControl } from './plugin'
+
+type UserConfig = z.infer<typeof Definition.schema.userConfig>
+type DataV1Type = z.infer<typeof Definition.schema.dataV1>
+type DataV2Type = z.infer<typeof Definition.schema.dataV2>
+type DataType = z.infer<typeof Definition.schema.data>
+type ValueType = z.infer<typeof Definition.schema.value>
+
+abstract class Definition<
+  RuntimeNode,
+  Config extends UserConfig = UserConfig,
+  InstanceType extends ControlInstance<any> = ControlInstance<any>,
```

Comment:
Parametrized by `InstanceType` because the current implementation of the control instance depends on Slate's `ReactEditor` type.